### PR TITLE
[keyboard] NEW smallkeyboard support ，Thanks

### PR DESCRIPTION
**Pull request by zhouqiong19840119**
## Description
small macro keyboard

## Types of Changes

- [x] Core
- [x] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/smallkeyboard/rules.mk [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+
+BLUETOOTH_ENABLE = no          # Enable Bluetooth
+AUDIO_ENABLE = no              # Audio output
+
+NO_USB_STARTUP_CHECK = no      # Disable initialization only when usb is plugged in
```
Comment (fauxpark) [inline keyboards/smallkeyboard/rules.mk:25]:
```suggestion
BLUETOOTH_ENABLE = no          # Enable Bluetooth
AUDIO_ENABLE = no              # Audio output

```

File: keyboards/smallkeyboard/smallkeyboard.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+#define LAYOUT_numpad_2x3( \
+    K00, K01, K02,\
+    K10, K11, K12 \
+) { \
+    { K00, K01, K02 }, \
+    { K10, K11, K12 }  \
+}
+
+#define LAYOUT_pad numpad_2x3
```
Comment (fauxpark) [inline keyboards/smallkeyboard/smallkeyboard.h:30]:
```suggestion
```

File: keyboards/smallkeyboard/readme.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+SMALLKEYBOARD
+
+Keyboard Maintainer: zhouqiong19840119
+Hardware Availability: https://github.com/zhouqiong19840119/MacroKeyboard-little/blob/main/Small%20(1).jpg
+Make example for this keyboard (after setting up your build environment):
+
+make smallkeyboard:default
+See the build environment setup and the make instructions for more information. Brand new to QMK? Start with our Complete Newbs Guide.
```
Comment (fauxpark) [inline keyboards/smallkeyboard/readme.md:8]:
```suggestion
# SMALLKEYBOARD

![](https://github.com/zhouqiong19840119/MacroKeyboard-little/blob/main/Small%20(1).jpg)

* Keyboard Maintainer: zhouqiong19840119
* Hardware Availability: 
* Make example for this keyboard (after setting up your build environment):

    make smallkeyboard:default

Flashing example for this keyboard:

    make smallkeyboard:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/smallkeyboard/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/* Copyright 2021 zhouqiong19840119
```
Comment (fauxpark) [inline keyboards/smallkeyboard/keymaps/via/keymap.c:1]: This keymap folder needs a `rules.mk` with:
```makefile
VIA_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/smallkeyboard/keymaps/via/keymap.c (hunk 4 above)
Comment (zhouqiong19840119) [reply keyboards/smallkeyboard/keymaps/via/keymap.c:1]: THANKS sir

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/12817/files).

File: keyboards/smallkeyboard/info.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+  "keyboard_name": "smallkeyboard",
+  "url": "",
+  "maintainer": "zhouqiong19840119",
+  "width": 3,
+  "height": 2,
```
Comment (fauxpark) [inline keyboards/smallkeyboard/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/smallkeyboard/config.h [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+/* disable these deprecated features by default */
+//#ifndef LINK_TIME_OPTIMIZATION_ENABLE
+//#    define NO_ACTION_MACRO
+//#    define NO_ACTION_FUNCTION
+//#endif
+
+#ifdef RGB_MATRIX_ENABLE
+#    define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
```
Comment (drashna) [inline keyboards/smallkeyboard/config.h:60]:
```suggestion
#    define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

File: keyboards/smallkeyboard/rules.mk [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/smallkeyboard/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Enable Bootmagic Lite
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/smallkeyboard/rules.mk [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/smallkeyboard/rules.mk:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**